lib/next_ls/runtime/sidecar.ex

```diff
@@ -31,4 +54,60 @@ defmodule NextLS.Runtime.Sidecar do
 
     {:noreply, state}
   end
+
+  defp filter_attributes(symbols) do
+    symbols
+    |> Enum.filter(&match?({:attribute, _, _, _}, &1))
+    |> Enum.reject(fn {_, "@" <> name, _, _} ->
+      Map.has_key?(Module.reserved_attributes(), String.to_atom(name))
+    end)
+  end
+
+  defp parse_symbols(file, module) do
+    ast = file |> File.read!() |> Code.string_to_quoted!(columns: true)
+
+    {_ast, %{symbols: symbols}} =
+      Macro.traverse(ast, %{modules: [], symbols: []}, &prewalk/2, &postwalk(&1, &2, module))
+
+    symbols
+  end
+
+  # add module name to modules stack on enter
+  defp prewalk({:defmodule, _, [{:__aliases__, _, modules} | _]} = ast, acc) do
+    modules_string =
+      modules
+      |> Enum.map(&Atom.to_string/1)
+      |> Enum.intersperse(".")
+      |> List.to_string()
```

Comment:
If I'm reading this correctly, I think you can use [Module.concat/1](https://hexdocs.pm/elixir/Module.html#concat/1) for this

Reply:
It'll be a alias tho (atom) rather than string, but I'm not sure they have to be strings for what you're doing.

Reply:
yeah, you are write. I don't need strings at all. Will rewrite and simplify it.